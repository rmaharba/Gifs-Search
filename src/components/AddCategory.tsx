import { ChangeEvent, FormEvent, useState } from "react";

type AddCategoryProps = {
  onAddCategory: (value: string) => void;
};

export const AddCategory = ({ onAddCategory }: AddCategoryProps) => {
  const [inputValue, setInputValue] = useState("");

  const onChangeInput = ({ target }: ChangeEvent<HTMLInputElement>) =>
    setInputValue(target.value);

  const onSubmitForm = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const inputValueFormated = inputValue.trim();
    if (inputValueFormated.length > 0) {
      onAddCategory(inputValueFormated);
      setInputValue("");
    }
  };

  return (
    <form aria-label="form" onSubmit={onSubmitForm}>
      <input
        type="text"
        value={inputValue}
        onChange={onChangeInput}
        placeholder="Buscar gifs"
      />
    </form>
  );
};
