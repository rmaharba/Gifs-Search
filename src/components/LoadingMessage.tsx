export const LoadingMessage = ({ isLoading }: { isLoading: boolean }) => {
  return <>{isLoading && <h2>Cargando...</h2>}</>;
};
