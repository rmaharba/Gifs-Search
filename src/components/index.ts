export * from "./GifGrid";
export * from "./AddCategory";
export * from "./GifGridItem";
export * from "./LoadingMessage";
