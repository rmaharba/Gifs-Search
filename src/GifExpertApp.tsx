import { useState } from "react";

import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Rick and Morty", "F1"]);

  const onAddCategory = (value: string) => {
    if (categories?.includes(value)) return;
    setCategories([value, ...categories]);
  };

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory onAddCategory={onAddCategory} />
      {categories?.map((item) => (
        <GifGrid key={item} category={item} />
      ))}
    </>
  );
};
