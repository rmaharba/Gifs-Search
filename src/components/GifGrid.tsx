import { useFetchGifs } from "../hooks";

import { GifGridItem } from "./GifGridItem";
import { LoadingMessage } from "./LoadingMessage";

type GifGridTypes = { category: string };

export type GifsArrayType = {
  id: string;
  title: string;
  url: string;
};

export const GifGrid = ({ category }: GifGridTypes) => {
  const { isLoading, images } = useFetchGifs({ category });

  return (
    <>
      <LoadingMessage isLoading={isLoading} />
      <h3>{category}</h3>
      <div className="card-grid">
        {images?.map((image) => (
          <GifGridItem key={image?.id} {...image} />
        ))}
      </div>
    </>
  );
};
