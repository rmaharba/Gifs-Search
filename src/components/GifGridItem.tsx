import { GifsArrayType } from "./GifGrid";

export const GifGridItem = ({ title, url }: GifsArrayType) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

// GifGridItem.propTypes = {
//   title: PropTypes.string.isRequired,
//   url: PropTypes.string.isRequired,
// };
