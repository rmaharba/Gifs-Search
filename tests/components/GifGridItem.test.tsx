import { render, screen } from "@testing-library/react";

import { GifGridItem } from "../../src/components";

describe("Test on <GifGridItem/>", () => {
  const title = "Saitama";
  const url = "https://example.com/image.png";

  test("should do match with snapshot", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show URL and ALT indicated", () => {
    render(<GifGridItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole<HTMLImageElement>("img").src).toBe(url);
    // expect(screen.getByRole<HTMLImageElement>("img").alt).toBe(title);

    const { alt, src } = screen.getByRole<HTMLImageElement>("img");
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test("should show title on the component", () => {
    render(<GifGridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
