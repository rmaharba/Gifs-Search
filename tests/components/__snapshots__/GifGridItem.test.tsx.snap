// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem/> should do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://example.com/image.png"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
